4_advanced_usage/40_node_properties.txt

```diff
@@ -0,0 +1,53 @@
+=== Node properties
+
+==== Defining properties
+
+Node properties can be found in the "properties" tab of each node in Rudder.
+
+Node properties can be modified using Rudder API, see http://www.rudder-project.org/rudder-api-doc/#api-Nodes-updateNodeProperties
+
+Currently a property is interpreted as a string value. So properties can be seen as a list of key=value pairs.
+
+==== Using properties
+
+You can use a node property almost everywhere in Rudder:
+- in directive parameters
+- in the technique editor
+- in your own CFEngine code
+
+To use a property, simply use the variable node.properties with the variable call syntax.
+
+Example with a string property named mykey:
+
+----
+
+${node.properties[mykey]}
+
+----
+
+
+==== Under the hood (CFEngine)
+
+On the server, one or more properties files are written for each node in the /var/rudder/share/<uuid>/rules/cfengine-community/properties.d/ directory.
+This directory is then synchronized by the agent with all other promise files.
+
+In CFEngine, properties are made available in the node.namespace container that contains the values.
```

Comment:
Can you replace "node.namespace" with "node.<namespace>" to illustrate that this is placeholder name that will change?